types/xregexp/index.d.ts

```diff
@@ -83,7 +84,7 @@ declare namespace OuterXRegExp {
     function uninstall(options: Object): void;
     function uninstall(options: string): void;
 
-    function union(patterns: string[], flags?: string): RegExp;
+    function union(patterns: string[] | RegExp[], flags?: string): RegExp;
```

Comment:
`string[] | RegExp[]` means you can pass `['hello', 'world']` or `[/hello/, /world/]` but not `['hello', /world/]`. `(string | RegExp)[]` allows `['hello', /world/]` which appears to be an allowed value based on the examples